src/main/java/uk/gov/hmcts/reform/iacasedocumentsapi/domain/templates/HearingRequirementsTemplate.java

```diff
@@ -51,15 +56,61 @@ public Map<String, Object> mapFieldValues(
 
         Optional<List<IdValue<WitnessDetails>>> witnessDetails = asylumCase.read(WITNESS_DETAILS);
         fieldValues.put(
-            "witnessDetails",
-            witnessDetails
-                .orElse(Collections.emptyList())
-                .stream()
-                .map(witnessIdValue ->
-                        ImmutableMap.of("witnessDetails", formatWitnessDetails(witnessIdValue.getValue())))
-                .collect(Collectors.toList())
+                "witnessDetails",
+                witnessDetails
+                        .orElse(Collections.emptyList())
+                        .stream()
+                        .map(witnessIdValue ->
+                                ImmutableMap.of("witnessDetails", formatWitnessDetails(witnessIdValue.getValue())))
+                        .collect(Collectors.toList())
         );
 
+        fieldValues.put("isAnyWitnessInterpreterRequired", asylumCase.read(IS_ANY_WITNESS_INTERPRETER_REQUIRED, YesOrNo.class).orElse(YesOrNo.NO));
```

Comment:
There are lots of logic being handled in this method, would it be better to extract some of them to private methods? I think it will help to separate the logics a bit and make them easier to understand.

Reply:
Done, extracted the logic into private two methods (one for witness and one for appellant)